packages/eslint-plugin/lib/rules/no-polymorphic-call.js

```diff
@@ -0,0 +1,52 @@
+"use strict"
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "disallow polymorphic function calls e.g.: 'array.slice()'",
+            category: "Possible Security Errors",
+            recommended: true,
+            url:
+                "https://github.com/endojs/endo/blob/master/packages/eslint-plugin/lib/rules/no-polymorphic-call.js",
+        },
+        type: "problem",
+        fixable: null,
+        schema: [],
+        supported: true,
+    },
+    create (context) {
+        return {
+          CallExpression(node) {
+            if (node.callee.type !== 'MemberExpression') {
+              return
+            }
+            const reportHint = prepareMemberExpressionHint(node.callee)
+            context.report(node, `Polymorphic call: "${reportHint}". May be vulnerable to corruption or trap`)
+          }
+        }
+    },
+}
+
+function prepareMemberExpressionHint (node) {
+  const { object, property, computed } = node
+  let objectHint
+  let propertyHint
+  if (object.type === 'Identifier') {
+    objectHint = object.name
+  } else if (object.type === 'MemberExpression') {
```

Comment:
Does it cover `a[b]`?

Reply:
This certainly did produce lint errors for `a[b]()` syntax.

Reply:
yes same member expression but with computed flag set to true